components/collapsed_reply_threads_tour/crt_welcome_tutorial_tip/crt_welcome_tutorial_tip.tsx

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+import React from 'react';
+import {FormattedMessage} from 'react-intl';
+
+import {useDispatch, useSelector} from 'react-redux';
+
+import FormattedMarkdownMessage from 'components/formatted_markdown_message.jsx';
+import TutorialTip from 'components/tutorial/tutorial_tip';
+import {Constants} from 'utils/constants';
+import {GlobalState} from 'types/store';
+import {close as closeLhs} from 'actions/views/lhs';
+import {getCurrentRelativeTeamUrl} from 'mattermost-redux/selectors/entities/teams';
+import {useMeasurePunchouts} from 'components/tutorial/tutorial_tip/hooks';
+import {browserHistory} from 'utils/browser_history';
+
+type Props = {
+    autoTour: boolean;
+};
+const CRTWelcomeTutorialTip = ({autoTour}: Props) => {
+    const teamUrl = useSelector((state: GlobalState) => getCurrentRelativeTeamUrl(state));
+    const nextUrl = `${teamUrl}/threads`;
+    const dispatch = useDispatch();
```

Comment:
hooks should always be placed first in a component

```suggestion
    const teamUrl = useSelector((state: GlobalState) => getCurrentRelativeTeamUrl(state));
    const dispatch = useDispatch();
    const nextUrl = `${teamUrl}/threads`;
```

Reply:
@michelengelen is there some reason behind it or just preference.

Reply:
The reason for this is well explained in this section of the docs: https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level

Reply:
But it just says top-level that doesn’t exactly mean declaring them first, as much I understand by top-level is hooks are supposed to be a direct child and not supposed to be nested inside anything else. But no worries will move it up. Thanks

Reply:
It's a bit of a personal rule to prevent errors. If you get accustomed to do that it is basically helping in structuring your components ... even if it is not needed at that specific component